File: JavaScript_Basic_04_2023_SolutionsOfProblems/05.LAB_PB-JS-While-Loop/P05.AccountBalance.js
function accountBalance(input) {
    let i = 0, curreentBalance = 0;
    while (input[i] !== "NoMoreMoney") {
        let money = Number(input[i]);
        if (money < 0) {
            console.log("Invalid operation!")
            break;
        }
        curreentBalance += money;
        console.log(`Increase: ${money.toFixed(2)}`);
        i++;
    }
    console.log(`Total: ${curreentBalance.toFixed(2)}`);
}
accountBalance(["120",
    "45.55",
    "-150"])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P05LAB_While_Loop/P05_AccountBalance.java
package P05LAB_While_Loop;

import java.util.Scanner;

public class P05_AccountBalance {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = "";
        double inputNumber = 0, sum = 0;
        while (true) {
            input = scanner.nextLine();
            if (input.equals("NoMoreMoney")) {
                break;
            }
            inputNumber = Double.parseDouble(input);
            if (inputNumber < 0) {
                System.out.println("Invalid operation!");
                break;
            }
            sum += inputNumber;
            System.out.printf("Increase: %.2f%n", inputNumber);
        }
        System.out.printf("Total: %.2f", sum);
    }
}
